src/modalities/models/gpt2/gpt2_model.py

```diff
@@ -133,9 +133,17 @@ def __init__(self, n_embd: int, n_head: int, seq_length_dim: int = -2, base_freq
         super().__init__()
         # this also holds when using TP, since n_embd is the total embedding size and
         # n_head is the number of heads globally
-        dim_model = n_embd // n_head
+        self.dim_model = n_embd // n_head
         self.seq_length_dim = seq_length_dim
-        inv_freq = 1.0 / (base_freq ** (torch.arange(0, dim_model, 2).float() / dim_model))
+        self.base_freq = base_freq
+
+        self.reset_parameters()
```

Comment:
do we have to call this explicitly in the constructor? Wouldn't our weight init routines call it?